microfactory/microfactory.go

```diff
@@ -69,6 +69,7 @@ var (
 
 	goToolDir = filepath.Join(runtime.GOROOT(), "pkg", "tool", runtime.GOOS+"_"+runtime.GOARCH)
 	goVersion = findGoVersion()
+	isGo18    = strings.Contains(goVersion, "go1.8")
 )
 
 func findGoVersion() string {
```

Comment:
I found out last time that you can't, since microfactory may be rebuilding itself after a Go upgrade, so we really want the go version that's in the GOROOT, not the one that previously existed and compiled microfactory.

In this case it's not that big of an issue, since you'd just lose some perf. The other use case of ensuring that the binaries got rebuilt when the go version changed was a bigger issue.

For the rebuilding case, any unique string works. For this -c case, the only time it would fail is if you used a development version that didn't have -c support (so a master/beta build before -c was added), since we're just blacklisting go1.8 (and require os.Executable so <1.8 isn't supported)